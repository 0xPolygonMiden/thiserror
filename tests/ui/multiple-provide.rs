#![feature(error_generic_member_access, provide_any)]

use thiserror::Error;
use std::any::Provider;
use std::error::Error;

// FIXME: this should work. https://github.com/dtolnay/thiserror/issues/185
#[derive(Error, Debug)]
#[error("...")]
struct MyError {
    #[source]
    #[backtrace]
    x: std::io::Error,
}

fn main() {
    let _: dyn Error;
    let _: dyn Provider;
}
