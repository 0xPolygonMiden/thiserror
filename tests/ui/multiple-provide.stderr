error[E0034]: multiple applicable items in scope
  --> tests/ui/multiple-provide.rs:13:5
   |
13 |     x: std::io::Error,
   |     ^ multiple `provide` found
   |
   = note: candidate #1 is defined in an impl of the trait `Provider` for the type `E`
   = note: candidate #2 is defined in an impl of the trait `std::error::Error` for the type `std::io::Error`
help: disambiguate the associated function for candidate #1
   |
13 |     Provider::provide(&x, Error): std::io::Error,
   |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
help: disambiguate the associated function for candidate #2
   |
13 |     std::error::Error::provide(&x, Error): std::io::Error,
   |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
